src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.DirectoryServices.Protocols;
+using System.Linq;
+using System.Net;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Authentication.Negotiate
+{
+    internal class LdapAdapter
+    {
+        private readonly string _distinguishedName;
+        private readonly LdapConnection _connection;
+        private readonly ILogger _logger;
+        private readonly LdapOptions _options;
+
+        public LdapAdapter(LdapOptions options, ILogger logger)
+        {
+            _logger = logger;
+            _options = options;
+            _distinguishedName = _options.Domain.Split('.').Select(name => $"dc={name}").Aggregate((a, b) => $"{a},{b}");
+
+            if (_options.LdapConnection == null)
```

Comment:
This should move to options. See 
https://github.com/dotnet/aspnetcore/blob/4ecde5f6b0d1aba59c7bd0646e20e7b16a78f767/src/Security/Authentication/OAuth/src/OAuthPostConfigureOptions.cs#L26